Tasks/DeployVisualStudioTestAgent/LocateTestAgentHelper.ps1

```diff
@@ -0,0 +1,82 @@
+function Locate-TestAgent($ProductVersion = "14.0")
+{
+    Write-Verbose "Locating Test Agent: $ProductVersion" -verbose
+    $testAgentDetails = @{}
+    $version = $null
+    $path = $null
+
+    #Find the latest version
+    $regPath = "HKLM:\SOFTWARE\Microsoft\DevDiv\vstf\Servicing"
+    if (-not (Test-Path $regPath))
+    {
+        $regPath = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\DevDiv\vstf\Servicing"
+    }
+    if (Test-Path $regPath)
+    {
+        $keys = Get-ChildItem $regPath | Where-Object {$_.GetSubKeyNames() -contains "testagentcore"}
+        $version = Get-SubKeysInFloatFormat $keys | Sort-Object -Descending | Select-Object -First 1
+        $path = $regPath + "\" + $version + "\" + "testagentcore"
```

Comment:
shouldnt we be using join-path here too?
registry key separators can also change per locale